src/wp-includes/class-wp-user-query.php

```diff
@@ -1010,6 +1037,54 @@ protected function parse_orderby( $orderby ) {
 		return $_orderby;
 	}
 
+	/**
+	 * Generate cache key.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @global wpdb $wpdb WordPress database abstraction object.
+	 *
+	 * @param array  $args Query arguments.
+	 * @param string $sql  SQL statement.
+	 *
+	 * @return string Cache key.
+	 */
+	protected function generate_cache_key( array $args, $sql ) {
+		global $wpdb;
+
+		// Replace wpdb placeholder in the SQL statement used by the cache key.
+		$sql = $wpdb->remove_placeholder_escape( $sql );
+
+		$key          = md5( $sql );
+		$last_changed = wp_cache_get_last_changed( 'users' );
+
+		if ( empty( $args['orderby'] ) ) {
+			// Default order is by 'user_login'.
+			$ordersby = array( 'user_login' => '' );
+		} elseif ( is_array( $args['orderby'] ) ) {
+			$ordersby = $args['orderby'];
+		} else {
+			// 'orderby' values may be a comma- or space-separated list.
+			$ordersby = preg_split( '/[,\s]+/', $args['orderby'] );
+		}
+
+		$blog_id = 0;
+		if ( isset( $args['blog_id'] ) ) {
+			$blog_id = absint( $args['blog_id'] );
+		}
+		if ( ( $args['has_published_posts'] && $blog_id ) || in_array( 'post_count', $ordersby, true ) ) {
+			if ( get_current_blog_id() !== $blog_id ) {
+				switch_to_blog( $blog_id );
+			}
+			$last_changed .= wp_cache_get_last_changed( 'posts' );
+			if ( get_current_blog_id() !== $blog_id ) {
+				restore_current_blog();
+			}
+		}
```

Comment:
```suggestion
		if ( ( $args['has_published_posts'] && $blog_id ) || in_array( 'post_count', $ordersby, true ) ) {
			$switch = false;
			if ( get_current_blog_id() !== $blog_id ) {
				$switch = true;
				switch_to_blog( $blog_id );
			}
			$last_changed .= wp_cache_get_last_changed( 'posts' );
			if ( $switch ) {
				restore_current_blog();
			}
		}
```

Similar to https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-admin/includes/ms.php#L71-L75 we can remove the duplicate condition.

Reply:
Fixed.